sketch/src/sketch_rnn/model.ts

```diff
@@ -0,0 +1,359 @@
+/**
+ * Core implementation for RNN-based Magenta sketch models such as SketchRNN.
+ *
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Imports
+ */
+import * as tf from '@tensorflow/tfjs-core';
+
+import * as support from '../core/sketch_support';
+
+export interface SketchRNNInfo {
+  max_seq_len: number;
+  mode: number;
+  name: string;
+  scale_factor: number;
+  version: number;
+}
+
+/**
+ * Main SketchRNN model class.
+ *
+ * Implementation of decoder model in https://arxiv.org/abs/1704.03477
+ */
+export class SketchRNN {
+  private checkpointURL: string;
+
+  private forgetBias: tf.Scalar;
+
+  private initialized: boolean;
+
+  public info: SketchRNNInfo;
+  public numUnits: number;
+
+  public pixelFactor: number;
+  public scaleFactor: number;
+
+  // raw weights and dimensions directly from JSON
+  private weights: Float32Array[];
+  private weightDims: number[][];
+
+  // TensorFlow.js weight matrices
+  private outputKernel: tf.Tensor2D;
+  private outputBias: tf.Tensor1D;
+  private lstmKernel: tf.Tensor2D;
+  private lstmBias: tf.Tensor1D;
+
+  private rawVars: tf.Tensor[];
+
+  private NMIXTURE = 20;
+
+  /**
+   * `SketchRNN` constructor.
+   *
+   * @param checkpointURL Path to the checkpoint directory.
+   */
+  constructor(checkpointURL: string) {
+    this.checkpointURL = checkpointURL;
+    this.initialized = false;
+  }
+
+  /**
+   * Returns true if model is intialized.
+   */
+  isInitialized() {
+    return this.initialized;
+  }
+
+  /**
+   * Instantiates class information inputs from the JSON model.
+   */
+  private instantiateFromJSON(info: SketchRNNInfo,
```

Comment:
add TODO to support new tfjs checkpoints

Reply:
sounds good.